### PR TITLE
Feature/rental reservation

### PR DESCRIPTION
**Pull request by mbanaszkiewicz**
I've decided to changed the relation between User and his Rentals and Reservations to unidirectional ManyToOne. Also I'll add tests later in the week when this implementation gets accepted.

### CODE REVIEW

File: backend/nokia-innovative-project/src/main/java/com/nokia/library/nokiainnovativeproject/entities/Rental.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+	@Future(message = "The return date should be future")
+	private Date returnDate;
+
+	@Setter
+	@ManyToOne(fetch = FetchType.LAZY)
+	@OnDelete(action = OnDeleteAction.CASCADE)
+	@JoinColumn(name = "user_id")
+	private User user;
+
+	@Setter
```
Comment (Dasiuss) [inline backend/nokia-innovative-project/src/main/java/com/nokia/library/nokiainnovativeproject/entities/Rental.java]: `@Setter` can be over class

File: backend/nokia-innovative-project/src/main/java/com/nokia/library/nokiainnovativeproject/entities/Rental.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+	@OnDelete(action = OnDeleteAction.CASCADE)
+	@JoinColumn(name = "book_catalog_number")
+	private Book book;
+
+	@Override
+	public boolean equals(Object o) {
+		if (this == o) return true;
+		if (!(o instanceof Rental )) return false;
+		return id != null && id.equals(((Rental) o).id);
+	}
```
Comment (Dasiuss) [inline backend/nokia-innovative-project/src/main/java/com/nokia/library/nokiainnovativeproject/entities/Rental.java]: equals without hashCode is asking for problems. you lombok for simplicity
on class: ```@EqualsAndHashCode(onlyExplicitlyIncluded = true)```
on id field: ```@EqualsAndHashCode.Include```

File: backend/nokia-innovative-project/src/main/java/com/nokia/library/nokiainnovativeproject/entities/Reservation.java [hunk 3]
```diff
@@ -15,33 +20,35 @@
...
+
+    @CreationTimestamp
+    @Setter(AccessLevel.NONE)
+    private Date reservationDate;
+
+    @NotNull(message = "The return date should be defined")
+    @Future(message = "The return date should be future")
+    private Date returnDate;
+
+    @Setter
```
Comment (Dasiuss) [inline backend/nokia-innovative-project/src/main/java/com/nokia/library/nokiainnovativeproject/entities/Reservation.java]: `@Setter` over class, `equals` should have `hashcode`

File: backend/nokia-innovative-project/src/main/java/com/nokia/library/nokiainnovativeproject/services/RentalService.java [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+    public Rental createRental(RentalDTO rentalDTO) {
+        ModelMapper mapper = new ModelMapper();
+        Rental rental = mapper.map(rentalDTO, Rental.class);
+        rental.setUser(userService.getUserById(rentalDTO.getUserId()));
+        rental.setBook(bookService.getBookById(rentalDTO.getBookId()));
+        return rentalRepository.save(rental);
+    }
+
+    public Object updateRental(Long id, RentalDTO rentalDTO) {
```
Comment (Dasiuss) [inline backend/nokia-innovative-project/src/main/java/com/nokia/library/nokiainnovativeproject/services/RentalService.java]: Object?

File: backend/nokia-innovative-project/src/main/java/com/nokia/library/nokiainnovativeproject/DTOs/RentalDTO.java [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    private Long userId;
+
+    @Getter
+    @NotNull(message = "Book ID is required.")
+    private Long bookId;
+
+    @NotNull(message = "The return date should be defined")
+    @Future(message = "The return date should be future")
+    @Getter
+    private Date returnDate;
```
[the remark this answers is not in the document]
Comment (MateuszPe) [reply backend/nokia-innovative-project/src/main/java/com/nokia/library/nokiainnovativeproject/DTOs/RentalDTO.java]: No, it shouldn't be.

**Comment by mbanaszkiewicz** [comment]
I've fixed the issues mentioned above. Is setting returnDate in constructor ok?

**Pull request merged**